File: OperationsApi/Features/TradeOrders/Models/TradeOrderModel.cs
using OperationsApi.Features.Operations.Models;
using OperationsApi.Features.TradeOrderTypes.Models;

namespace OperationsApi.Features.TradeOrders.Models
{
    public class TradeOrderModel
    {
        public int ID { get; set; }
        public int OperationID { get; set; }
        public decimal Amount { get; set; }
        public int TradeOrderTypeID { get; set; }
        public TradeOrderTypeModel TradeOrderType { get; set; }
        public OperationModel Operation { get; set; }
    }
}

File: client/operations/src/shared/models/trade-order.model.ts
import { OperationModel } from "./operation.model";
import { TradeOrderTypeModel } from "./trade-order-type.model";

export interface TradeOrderModel {
    id: number;
    operationid: number;
    amount: number;
    tradeOrderTypeID: number;
    tradeOrderType: TradeOrderTypeModel;
    operation: OperationModel;
}